packages/contentful--app-scripts/src/create-app-definition/build-app-parameter-settings.test.ts

```diff
@@ -0,0 +1,206 @@
+import sinon from 'sinon';
+import inquirer from 'inquirer';
+import { buildAppParameterSettings } from './build-app-parameter-settings';
+
+describe('buildAppParameterSettings', async () => {
+  let chai: Chai.ChaiStatic;
+  let promptStub: sinon.SinonStub;
+
+  beforeEach(async () => {
+    promptStub = sinon.stub(inquirer, 'prompt');
+    chai = await import('chai');
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it('should add instance parameters correctly', async () => {
+    promptStub.onCall(0).resolves({
+      instanceOrInstallation: 'instance',
+      name: 'param1',
+      id: 'param1',
+      type: 'Symbol',
+      required: false,
+    });
+    promptStub.onCall(1).resolves({ addAnother: false });
+
+    const parameters = await buildAppParameterSettings();
+
+    chai.expect(parameters.instance).to.deep.equal([
+      {
+        name: 'param1',
+        id: 'param1',
+        type: 'Symbol',
+        required: false,
+        default: undefined,
```

Comment:
are all these `x: undefined` necessary?

Reply:
yeah this is weird.. chai deep equal should work in theory without the undefineds but the tests fail if I omit them